### PR TITLE
Adding external authentication capability when using oracledb driver

### PR DESCRIPTION
**Pull request by nixgadget**
Adding external authentication capability as mentioned in https://github.com/oracle/node-oracledb/blob/master/doc/api.md#extauth

### CODE REVIEW

**Comment by nixgadget** [comment]
What does it take to merge these changes upstream ?

File: src/dialects/oracledb/index.js [hunk 1]
```diff
@@ -64,15 +64,27 @@ Client_Oracledb.prototype.acquireRawConnection = function() {
...
+    // If external authentication dont have to worry about username/password
+    if (client.connectionSettings.externalAuth) {
+      oracleDbConfig.externalAuth = client.connectionSettings.externalAuth;
+    } else {
+      oracleDbConfig.user = client.connectionSettings.user;
+      oracleDbConfig.password = client.connectionSettings.password;
     }
+
+    // In the case of external authentication connection string will be given
+    if (client.connectionSettings.connectString){
```
Comment (elhigu) [inline src/dialects/oracledb/index.js]: Here the original style
```js
oracleDbConfig.connectString = client.connectionSettings.connectString ||
  (client.connectionSettings.host + '/' + client.connectionSettings.database);
```
is easier to read and understand.

File: src/dialects/oracledb/index.js [hunk 2]
```diff
@@ -64,15 +64,27 @@ Client_Oracledb.prototype.acquireRawConnection = function() {
...
 
-    const oracleDbConfig = {
-      user: client.connectionSettings.user,
-      password: client.connectionSettings.password,
-      connectString: client.connectionSettings.connectString ||
-        (client.connectionSettings.host + '/' + client.connectionSettings.database)
+    const oracleDbConfig = {};
+
+    // If external authentication dont have to worry about username/password
+    if (client.connectionSettings.externalAuth) {
```
Comment (elhigu) [inline src/dialects/oracledb/index.js]: This should be the same but more readable:

```js
const oracleDbConfig = client.connectionSettings.externalAuth ?
  { externalAuth : client.connectionSettings.externalAuth } :
  {
     user : client.connectionSettings.user,
     password: client.connectionSettings.password
  };
```

**Comment by xudingding978** [comment]
Hi, I did the changes you mentioned in the previous comment.
And the tests are failing at the "npm run babel" step, and mysql part.

**Comment by elhigu** [comment]
Thanks. Yeah, all builds has been annoyingly semi-random-pretty-much-always failing for some time. I'll look to the random fails and afterwards I'll review this one again.

File: src/dialects/oracledb/index.js [hunk 3]
```diff
@@ -65,15 +65,23 @@ Client_Oracledb.prototype.acquireRawConnection = function() {
...
-    const oracleDbConfig = {
-      user: client.connectionSettings.user,
-      password: client.connectionSettings.password,
-      connectString: client.connectionSettings.connectString ||
-        (client.connectionSettings.host + '/' + client.connectionSettings.database)
-    }
+    // If external authentication dont have to worry about username/password and
+    // if not need to set the username and password
+    const oracleDbConfig = client.connectionSettings.externalAuth ?
+        { externalAuth : client.connectionSettings.externalAuth } :
```
Comment (elhigu) [inline src/dialects/oracledb/index.js]: 2 spaces too wide indentation

**Comment by elhigu** [comment]
Looks good now, thanks! I added separate issue to documentation branch about documenting all the supported oracledb driver parameters.

**Pull request merged**